laws/src/main/scala/org/http4s/laws/discipline/arbitrary.scala

```diff
@@ -815,6 +815,24 @@ private[discipline] trait ArbitraryInstances { this: ArbitraryInstancesBinCompat
     } yield Uri(scheme, authority, path, query, fragment)
   }
 
+  /** Creates an `Arbitrary[Url]` with checking that a URI is converting to String
+    * and back to URI safely.
+    * Use this Arbitrary with cautions - it may lead to tests performance degradation.
+    */
+  def createArbitraryUri: Arbitrary[Uri] =
```

Comment:
Isn't a non-implicit `Arbitrary` just a `Gen`?

Reply:
Not necessarily. If you make it a `Gen`, it becomes annoying to use it:
https://github.com/http4s/http4s/blob/6ff6a74f877a81cd9fb59eaf2fa2d2412d4923e1/play-json/src/test/scala/org/http4s/play/PlaySuite.scala#L67

Reply:
IIUC, the reason to keep it non-implicit is not so it doesn't get used implicitly, but so that doing so is opt-in.

Reply:
I agree with @armanbilge. If we would use a `Gen` here then we need to wrap it into an `Arbitrary` everywhere. And at the moment don't see how we can reuse it. As well as we shouldn't have this `Arbitrary` as an implicit value in the public API because of its 'hacking' nature, I believe.

Reply:
But can't you call a `Gen` as `forAll(genUri) { uri => ... }` and not have to make it locally implicit?  

I strongly agree with not making it implicit in the `arbitrary` object because it's hacky.  Having agreed on that, I don't think the `Gen` vs. `Arbitrary` matters much either way.

Reply:
@rossabaker oh, sorry, I didn't get it the first time. Your suggestion is pretty nice. Then we will not deal with that tricky implicit `Arbitrary[Uri]` at all.